homeassistant/components/co2signal/sensor.py

```diff
@@ -114,6 +114,11 @@ def __init__(
             f"{coordinator.entry_id}_{description.unique_id or description.key}"
         )
 
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._description.key in self.coordinator.data["data"]
```

Comment:
This implements the coordinator entity, so the parent availability matters

```suggestion
        return super().available and self._description.key in self.coordinator.data["data"]
```